ElixrTalentConnect/SignUP/SignUpViewController.swift

```diff
@@ -0,0 +1,108 @@
+    @IBOutlet weak var backButton: UIButton!
+    @IBOutlet weak var SignUpTableview: UITableView!
+    @IBOutlet weak var promptSignup: UILabel!
+    
+    /// Variable Declaration.
+    var Dataheading :[String]?
+    var firstCellPlaceholder:String?
+    let cellTypes: [DataModel] = [.fullName, .emailAdress, .password, .confirmPassword]
+    let textToChangeColor = " Sign in."
+    let userInfoPrompt = "Already have an account? Sign in."
+    let viewModalInstance = SignUpViewModel()
+    
+    /// View life Cycle.
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        backButton.setImage(UIImage(systemName: "arrow.backward"), for: .highlighted)
+        backButton.tintColor = UIColor.gray
+        SigupPromptSetup()
+        self.view.endEditing(true)
+        
+    }
+    
+    /// SignIn actiion
+    /// - Parameter sender: UIButton
+    @IBAction func SignUpTapped(_ sender: UIButton) {
+        authenticationOnSignup()
+    }
+
+    ///@IBAction to set up return button to the rootview controller.
+    /// - Parameter sender: UIButton
+    @IBAction func returnAction(_ sender: UIButton) {
+        navigationController?.popViewController(animated: true)
+    }
+    
+    /// To  present a label that have attributed text field and tap gesture.
+    func SigupPromptSetup(){
+        let range = (userInfoPrompt as NSString).range(of: textToChangeColor)
+        let attributedText = NSMutableAttributedString.init(string: userInfoPrompt)
+        attributedText.addAttribute(NSAttributedString.Key.foregroundColor, value: UIColor(red: 228/255, green: 118/255, blue: 75/255, alpha: 1.0) , range: range)
+        promptSignup.attributedText = attributedText
+        let signInGesture = UITapGestureRecognizer(target: self, action:#selector(signUpAction(gesture: ) ))
+        promptSignup.isUserInteractionEnabled = true
+        promptSignup.addGestureRecognizer(signInGesture)
+    }
+    
+    /// Action of tap gesture.
+    /// - Parameter gesture: UItapgesture.
+    @objc func  signUpAction(gesture: UITapGestureRecognizer ){
+        let termsRange = (userInfoPrompt as NSString).range(of: self.textToChangeColor)
+        promptSignup.isUserInteractionEnabled = true
+        if gesture.didTapAttributedTextInLabel(label: promptSignup, inRange: termsRange) {
+            print("tapped")
+            guard let rootView = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "LoginViewController") as? LoginViewController
+            else {
+                return
+            }
+            navigationController?.pushViewController(rootView, animated: true)
+        }
+    }
+    
+    /// Function to navigate on validating some crirteria on the text fields, setting a reference from viewModal.
+    func authenticationOnSignup(){
+        let fullNameCell = SignUpTableview.cellForRow(at: IndexPath(row: viewModalInstance.cellTypes.firstIndex(of: .fullName) ?? 0, section: 0)) as? ProfileDisplayCell
```

Comment:
method naming convention and do we really need to handle like this?? can we handle it using the enum created ??